draft-ietf-ppm-dap.md

```diff
@@ -1295,44 +1299,70 @@ struct {
 } HpkeConfig;
 
 opaque HpkePublicKey<0..2^16-1>;
-uint16 HpkeAeadId; /* Defined in [HPKE] */
-uint16 HpkeKemId;  /* Defined in [HPKE] */
-uint16 HpkeKdfId;  /* Defined in [HPKE] */
+uint16 HpkeAeadId;
+uint16 HpkeKemId;
+uint16 HpkeKdfId;
 ~~~
 
-Aggregators MUST allocate distinct id values for each `HpkeConfig` in an
+The possible values for `HpkeAeadId`, `HpkeKemId` and `HpkeKdfId` are as defined
+in {{!HPKE, Section 7}}.
+
+Aggregators MUST allocate distinct `id` values for each `HpkeConfig` in an
 `HpkeConfigList`.
 
-The Client MUST abort if any of the following happen for any HPKE config
-request:
+The Client MUST abort if:
 
-* the GET request did not return a valid HPKE config list;
+* the response is not a valid HPKE config list;
 * the HPKE config list is empty; or
-* no HPKE config advertised by the Aggregator specifies a supported a KEM, KDF,
-  or AEAD algorithm triple.
-
-Aggregators SHOULD use HTTP caching to permit client-side caching of this
-resource {{!RFC5861}}. Aggregators SHOULD favor long cache lifetimes to avoid
-frequent cache revalidation, e.g., on the order of days. Aggregators can control
-this cached lifetime with the Cache-Control header, as in this example:
-
-~~~ http-message
-  Cache-Control: max-age=86400
+* no HPKE config advertised by the Aggregator specifies a supported KEM, KDF and
+  AEAD algorithm triple.
+
+Aggregators SHOULD use caching to permit client-side caching of this resource
+{{!RFC9111}}. Aggregators SHOULD favor long cache lifetimes to avoid frequent
+cache revalidation, e.g., on the order of days. Aggregators can control this
+cached lifetime with the Cache-Control header.
+
+Servers should choose a Cache-Control value appropriate to the lifetime of their
+keys. Clients SHOULD follow the usual caching semantics {{!RFC9111}} for HPKE
+configurations.
+
+Long cache lifetimes may result in Clients using stale HPKE configurations.
```

Comment:
Unrelated to this PR, but it's not clear to me how a long cache lifetime would result in stale configs being used.

Reply:
The case that comes to mind is if an aggregator rotates its HPKE config(s) ahead of the lifetime implied by `Cache-Control`. But that sounds like dubious practice on the part of an Aggregator: either the rotation period should be very close to the cache lifetime, or the key was rotated early because of a known compromise, in which case the server shouldn't honor messages encrypted to it anymore.

I'd be OK with deleting this sentence. What really matters is the following one about how long to accept reports with old keys.

Reply:
Agreed. Let's delete the sentence.